src/main/java/com/thebois/listeners/events/OnClickEvent.java

```diff
@@ -0,0 +1,25 @@
+package com.thebois.listeners.events;
+
+/**
+ * Sends out information about a value when notifier has been clicked.
```

Comment:
>when notifier has been clicked

What's a notifier? I suppose you're avoiding using "button" since it is more generic than that and can be used for things other than buttons. I'd rephrase the entire javadoc to not describe it as if it is sending out any information, because this is actually just a data-carrier object, it doesn't send anything.

"Contains information about a value that has been clicked." or something similar, better describes it imo.